data/users/externalldap.py

```diff
@@ -147,11 +147,20 @@ def _get_ldap_referral_dn(self, referral_exception):
         referral_dn = referral_uri[len("ldap:///") :]
         return referral_dn
 
+    def _clean_user_filter(self, query):
+        if not query:
+            return None
+
+        if query[0] == "(":
+            return query
```

Comment:
Please add `assert query[-1] == ')'`  here

Reply:
it is now wrapped in parens if the string does not start _and_ end with one